docs: filling the gaps of self hosted guide (#863)
kamilkisiela/graphql-hive: packages/web/docs/src/pages/self-hosting/get-started.mdx

First download the `docker-compose.community.yml` file from the
[GitHub repository](https://github.com/kamilkisiela/graphql-hive/blob/main/docker-compose.community.yml)

> using `wget`. This's gonna download the latest version available.

Reviewer: ```suggestion
using `wget`. This's going to download the latest version available.
```

<Callout type="info">

>   Note: make sure you've docker installed to be able to continue with the setup process.

Reviewer 2: nit: `Docker` instead of `docker`

>   Note: make sure you've docker installed to be able to continue with the setup process.

Author: ```suggestion
  Note: make sure you've Docker installed to be able to continue with the setup process.
```

</Callout>

> 

Author: ```suggestion
> You can also download the file directly from GitHub, if you don't have wget installed using the link below
```

```bash
wget https://raw.githubusercontent.com/kamilkisiela/graphql-hive/main/docker-compose.community.yml

> ```

Reviewer 2: You can also mention here: You can also download the file directly from GitHub, if you don't have `wget` installed.

```bash
export DOCKER_REGISTRY="ghcr.io/kamilkisiela/graphql-hive/"

> export DOCKER_TAG=":latest" # Pin this to an exact commit hash from our `main` branch

Reviewer 2: Mention here that we publish a Docker image per each commit on `main`, and for PRs.

> export DOCKER_TAG=":latest" # Pin this to an exact commit hash from our `main` branch

Author: ```suggestion
export DOCKER_TAG=":latest" # Pin this to an exact commit hash from our `main` branch, we publish a Docker image per each commit on main, and for PRs.
```

</Callout>

After setting the environment variables, pull the required images for GraphQL Hive services. This's
gonna take some time if you're doing it for the first time.

```bash
docker compose -f docker-compose.community.yml pull
```

After it's done, you can start the GraphQL Hive services using `docker compose`.

```bash
docker compose -f docker-compose.community.yml up

> # or of you embedded the environment variables

Author: ```suggestion
# or of you embedded the environment variables into your docker compose file and called it `docker-compose.with-env.yml`
```

> # or of you embedded the environment variables

Author: ```suggestion
# or of you embedded the environment variables into your docker compose file and called it `docker-compose.with-env.yml`
```

> docker compose -f docker-compose.with-env.yml up

Reviewer 2: I don't see `docker-compose.with-env.yml` mentioned above?

Author: oh so this's a modified version for people who included their env vars in the docker compose but I should indeed clarify that above the line.

```

Wait until all the services are up and running.

Congratulations 🥳, you just started your own GraphQL Hive instance.

You'll notice the folder named `.hive` at your root directory that has been mounted from docker

> representing the volumes for the different storages used by Hive services like `postgres` and

Reviewer 2: Please mention here that this directory is acting as a volume for storing the data from Hive dependencies: database and storage. Also mention that deleting this directly will end up with a loss of data.

> `minio`, etc.. So yes data are currently local but we'll discuss that further more in the

Reviewer 2: nit: `etc...` and not `etc..`

> `minio`, etc.. So yes data are currently local but we'll discuss that further more in the

Author: ```suggestion
`minio`, etc... So yes data are currently local but we'll discuss that further more in the
```

`Next Steps` section.

> 

Author: ```suggestion

> Note: deleting this directory will end up with a loss of data, as this directory is acting as a volume for storing the data from Hive dependencies: database and storage

```

endpoint, by creating a `hive.json` file at the root of your directory. Then specify the `registry`
key to point it our local registry instead of the cloud version one and the value is gonna be
`http://localhost:8082/graphql` where its running.

Additionally the schemas registry only allows trusted sources to be able to push to them, so we have
to create an access token by heading back to our project page, specifically on the target(in this
guide it's `development`) then go to the settings tab, if you scroll donw a bit you're gonna see the
`Tokens` section, click the `Generate New Token` button, give it a name then we'll have to set some
permissions for that token, so expand the `All targets` section then give the `Registry` a
`Read & Write` access. Click the `Generate Token` button, then copy the token value. Head back to
the `hive.json` config file then add a new key `token` and paste the value from your clipboard.

Now we're all setup to use the GraphQL Hive CLI against our Setup!

We can publish our first schema by using:

```bash
pnpm hive schema:publish ./schema.graphql
```

Go to the `Schema` tab in your project target and you're gonna see your latest published schema.

> ## Next Steps

Reviewer 2: Maybe we should have a section here to describe all the exposed ports and what's their purpose?

Author: oh yeah, definitely. I was confused myself by the ports available and what their purposes are

key to point it our local registry instead of the cloud version one and the value is gonna be
`http://localhost:8082/graphql` where its running.

Additionally the schemas registry only allows trusted sources to be able to push to them, so we have
to create an access token by heading back to our project page, specifically on the target(in this
guide it's `development`) then go to the settings tab, if you scroll donw a bit you're gonna see the
`Tokens` section, click the `Generate New Token` button, give it a name then we'll have to set some
permissions for that token, so expand the `All targets` section then give the `Registry` a
`Read & Write` access. Click the `Generate Token` button, then copy the token value. Head back to
the `hive.json` config file then add a new key `token` and paste the value from your clipboard.

Now we're all setup to use the GraphQL Hive CLI against our Setup!

We can publish our first schema by using:

```bash
pnpm hive schema:publish ./schema.graphql
```

Go to the `Schema` tab in your project target and you're gonna see your latest published schema.

## Next Steps

> 

Author: what do you think @dotansimha? 👀 

```suggestion
## Bounded Ports Reference

| Port | Purpose                      |
| ---- | ---------------------------- |
| 8080 | Hive's Main App UI           |
| 8081 | The Usage Reporting Endpoint |
| 8082 | The GraphQL API              |
| 8083 | The Artifacts Bucket         |

```

</Callout>

After setting the environment variables, pull the required images for GraphQL Hive services. This's

> gonna take some time if you're doing it for the first time.

Reviewer: ```suggestion
going to take some time if you're doing it for the first time.
```

```

Wait until all the services are up and running.

Congratulations 🥳, you just started your own GraphQL Hive instance.

You'll notice the folder named `.hive` at your root directory that has been mounted from docker
representing the volumes for the different storages used by Hive services like `postgres` and
`minio`, etc... So yes data are currently local but we'll discuss that further more in the
`Next Steps` section.

> Note: deleting this directory will end up with a loss of data, as this directory is acting as a volume for storing the data from Hive dependencies: database and storage

## Testing GraphQL Hive

Visit `http://localhost:8080` in your browser and start using GraphQL Hive! The usage reporting
endpoint is bound to `http://localhost:8081`. The GraphQL API is bound to `http://localhost:8082`.
The artifacts bucket is bound to `http://localhost:8083`.

Firstly, you can head to `http://localhost:8080` in your browser and start using GraphQL Hive!
You'll need to Sign Up for an account by inputting your email and password. And once you do you'll
see your personal organization that has been automatically created for you by default.

### Creating first project and target

> Now that we have GraphQL Hive setup locally, signed up and logged into our dashboard. We can start
> creating our first project which we're gonna push graphql schemas to.
> 
> So you're gonna click on the `Create a Project` button on the top right, give it a name and choose
> your project type. You're gonna see all sorts of types `Single` which is the most common and
> there're also `Federation` and `Stitching` but what's also nice is that you can define your own
> custom endpoints for schema validation and schema building. I'm gonna go with `Single` but you can
> choose what best suits your needs. Then click `Create Project`, you're gonna be redirected to your
> project page where you're gonna see 3 listed targets created for you by default:
> 
> - `development`
> - `production`
> - `staging`
> 
> We're gonna use the `development` target in this guide and as soon as you click it you'll see this
> message:
> 
> > Hive is waiting for your first schema.
> >
> > You can publish a schema with Hive CLI and Hive Client
> 
> If you already have a Nodejs project with a graphql schema that's great you can skip over the next
> step, but if not so we're gonna initialize a new package in our directory, you can use your
> preferred package manager to do so:

Reviewer: ```suggestion
Now that we have GraphQL Hive setup locally, signed up and logged into our dashboard. We can start
creating our first project which we're going to push graphql schemas to.

So you're going to click on the `Create a Project` button on the top right, give it a name and choose
your project type. You're going to see all sorts of types `Single` which is the most common and
there're also `Federation` and `Stitching` but what's also nice is that you can define your own
custom endpoints for schema validation and schema building. I'm going to go with `Single` but you can
choose what best suits your needs. Then click `Create Project`, you're going to be redirected to your
project page where you're going to see 3 listed targets created for you by default:

- `development`
- `production`
- `staging`

We're going to use the `development` target in this guide and as soon as you click it you'll see this
message:

> Hive is waiting for your first schema.
>
> You can publish a schema with Hive CLI and Hive Client

If you already have a Nodejs project with a graphql schema that's great you can skip over the next
step, but if not so we're going to initialize a new package in our directory, you can use your
preferred package manager to do so:
```

Reviewer: basically all `gonna` into `going to` :) Just to keep it in the same style as the rest of the documentation

Author: lol

```bash
# you can do the same with npm or yarn
pnpm init
```

Then create a `schema.graphql` file wherever you want, you can put it at the root for example with
any simple graphql schema like the following:

```graphql
type Query {
  hello: String!
}
```

> You also need to have a git repository initilaized for your project, because Hive automatically uses
> the last commit hash to determine who the author is and ties the schema with the commit hash. So
> we're gonna do that.
> 
> ```bash
> git init
> ```
> 
> Make sure you exclude the following in your `.gitignore` file, because you don't wanna push those in
> your commits:
> 
> ```
> node_modules
> .hive
> hive.json
> .env
> ```
> 
> Make sure to at least have a single commit so that Hive can associate the schema with it.
> 
> ```bash
> git add .
> git commit -m"init"
> ```
> 
> Now that you've your Nodejs project setup, start by installing the GraphQL Hive CLI which we're
> gonna use to interact with Hive's server.

Reviewer: ```suggestion
You also need to have a git repository initilaized for your project, because Hive automatically uses
the last commit hash to determine who the author is and ties the schema with the commit hash. So
we're going to do that.

```bash
git init
```

Make sure you exclude the following in your `.gitignore` file, because you don't wanna push those in
your commits:

```
node_modules
.hive
hive.json
.env
```

Make sure to at least have a single commit so that Hive can associate the schema with it.

```bash
git add .
git commit -m"init"
```

Now that you've your Nodejs project setup, start by installing the GraphQL Hive CLI which we're
going to use to interact with Hive's server.
```
